File: rb101/rb101-rb109-small_problems/easy6/triangle.rb
def triangle(n)
  space_counter = n - 1

  while space_counter >= 0
    puts "#{' ' * space_counter}#{'*' * (n - space_counter)}"
    space_counter -= 1
  end
end

triangle(5)
triangle(9)

# ALTERNATIVE SOLUTION:

# def triangle(num)
#   spaces = num - 1
#   stars = 1

#   num.times do |n|
#     puts (' ' * spaces) + ('*' * stars)
#     spaces -= 1
#     stars += 1
#   end
# end

# FURTHER EXPLORATION:

# Upside down solution:

# def triangle(n)
#   n.times do |spaces|
#     puts "#{' ' * spaces}#{'*' * (n - spaces)}"
#   end
# end

# Modified solution for every corner on grid:

# def print_bl_triangle(num)
#   num.times do |stars|
#     puts '*' * (stars + 1)
#   end
# end

# def print_br_triangle(num)
#   spaces = num - 1
#   num.times do |_|
#     puts "#{' ' * spaces}#{'*' * (num - spaces)}"
#     spaces -= 1
#   end
# end

# def print_tl_triangle(num)
#   num.times do |spaces|
#     puts "#{'*' * (num - spaces)}#{' ' * spaces}"
#   end
# end

# def print_tr_triangle(num)
#   num.times do |spaces|
#     puts "#{' ' * spaces}#{'*' * (num - spaces)}"
#   end
# end

# def triangle(num, angle_corner)
#   case angle_corner
#   when 'BL'
#     print_bl_triangle(num)
#   when 'BR'
#     print_br_triangle(num)
#   when 'TL'
#     print_tl_triangle(num)
#   when 'TR'
#     print_tr_triangle(num)
#   end
# end

File: js210/js210-small_problems/easy2/triangle.js
function triangle(height) {
  for (let starCount = 1; starCount <= height; starCount++) {
    let spaces = ' '.repeat(height - starCount);
    let stars = '*'.repeat(starCount);
    console.log(spaces + stars);
  }
}

triangle(5);
triangle(9);

// ALTERNATIVE SOLUTION:

// function triangle(height) {
//   let stars = 1;
//   let spaces = height - 1;

//   for (let i = 0; i < height; i += 1) {
//     console.log(repeat(' ', spaces) + repeat('*', stars));
//     stars += 1;
//     spaces -= 1;
//   }
// }

// function repeat(char, count) {
//   let repeated = '';

//   for (let i = 0; i < count; i += 1) {
//     repeated += char;
//   }

//   return repeated;
// }
